src/lib.rs

```diff
@@ -457,10 +385,24 @@ macro_rules! animator_impl {
 pub struct Animator<T: Component> {
     /// Control if this animation is played or not.
     pub state: AnimatorState,
-    tweenable: Option<BoxedTweenable<T>>,
+    tweenable: BoxedTweenable<T>,
     speed: f32,
 }
 
+impl<T: Component> Deref for Animator<T> {
```

Comment:
I don't really like this. Yes, it makes it convenient that we can call some methods of the tweenable from the `Animator` controlling it. But they're fundamentally different objects. And as far as I know it's considered bad practice implementing `Deref` for something that's not just a kind of smart pointer:

> Implementing Deref for smart pointers makes accessing the data behind them convenient, which is why they implement Deref. On the other hand, the rules regarding Deref and [DerefMut](https://doc.rust-lang.org/std/ops/trait.DerefMut.html) were designed specifically to accommodate smart pointers. Because of this, **Deref should only be implemented for smart pointers to avoid confusion.**

https://doc.rust-lang.org/std/ops/trait.Deref.html

Reply:
Yeah, I think that makes sense: I added back the tweenable/tweenable_mut functions.